ddtrace/contrib/botocore/patch.py

```diff
@@ -2,29 +2,56 @@
 Trace queries to aws api done via botocore client
 """
 # 3p
+import base64
+import json
 from ddtrace.vendor import wrapt
 from ddtrace import config
 import botocore.client
-import base64
-import json
 
 # project
 from ...constants import ANALYTICS_SAMPLE_RATE_KEY, SPAN_MEASURED_KEY
 from ...ext import SpanTypes, http, aws
-from ...internal.logger import get_logger
 from ...pin import Pin
-from ...propagation.http import HTTPPropagator
 from ...utils.formats import deep_getattr
 from ...utils.wrappers import unwrap
+from ...internal.logger import get_logger
+from ...propagation.http import HTTPPropagator
 
 # Original botocore client class
 _Botocore_client = botocore.client.BaseClient
 
 ARGS_NAME = ('action', 'params', 'path', 'verb')
 TRACED_ARGS = ['params', 'path', 'verb']
 
-propagator = HTTPPropagator()
 log = get_logger(__name__)
+propagator = HTTPPropagator()
+
+
+def inject_trace_data_to_message_attributes(trace_data, entry):
+    if 'MessageAttributes' not in entry:
+        entry['MessageAttributes'] = {}
+    if len(entry['MessageAttributes']) < 10:
```

Comment:
Why does this condition exist, is there a limit on the number of attributes? Would be great to have a comment above this line with a bit of an explanation :)

Reply:
Might also be worth having a debug log for when this happens because we won't be adding the distributed tracing headers

Reply:
Added a comment. More info here https://aws.amazon.com/sqs/faqs/#:~:text=support%20message%20metadata%3F-,Yes.,up%20to%2010%20metadata%20attributes.